### PR TITLE
latency(cdc): 2 phase scheduling processor bug fix

### PR DESCRIPTION
**Pull request by 3AceShowHand**
### What problem does this PR solve?

Issue Number: ref #4757

#### Tests 

 - No code

### Release note 

```release-note
`None`
```

### CODE REVIEW

File: cdc/scheduler/internal/tp/schedulepb/table_schedule.pb.go [hunk 1]
```diff
@@ -2473,7 +2473,7 @@ func (m *AddTableResponse) Unmarshal(dAtA []byte) error {
 		switch fieldNum {
 		case 1:
 			if wireType != 2 {
-				return fmt.Errorf("proto: wrong wireType = %d for field Status", wireType)
```
Comment (overvenus) [inline cdc/scheduler/internal/tp/schedulepb/table_schedule.pb.go:2476]: Please do not change generated files manually.

File: cdc/processor/processor.go [hunk 2]
```diff
@@ -268,19 +268,9 @@ func (p *processor) IsAddTableFinished(ctx context.Context, tableID model.TableI
...
-		// how about just check status is `TableStateReplicating`.
-		if table.CheckpointTs() < localCheckpointTs || localCheckpointTs < globalCheckpointTs {
-			return false
-		}
-		if table.ResolvedTs() < localResolvedTs ||
-			localResolvedTs < globalResolvedTs {
-			return false
-		}
-		return true
+		return table.State() == pipeline.TableStateReplicating
```
Comment (overvenus) [inline cdc/processor/processor.go:273]: Could you add some comments?

File: cdc/processor/pipeline/sink.go [hunk 3]
```diff
@@ -240,16 +241,18 @@ func splitUpdateEvent(updateEvent *model.PolymorphicEvent) (*model.PolymorphicEv
...
+	if n.state.Load() == TableStateStopped {
 		return false, cerror.ErrTableProcessorStoppedSafely.GenWithStackByArgs()
 	}
 	switch msg.Tp {
 	case pmessage.MessageTypePolymorphicEvent:
 		event := msg.PolymorphicEvent
+		if n.state.Load() != TableStateReplicating {
+			log.Panic("table not in replicating state",
+				zap.Int64("tableID", n.tableID),
+				zap.Any("state", n.state.Load()))
```
Comment (overvenus) [inline cdc/processor/pipeline/sink.go:253]: Could you add changefeed fields?

**Comment by overvenus** [comment]
Test TestTableExecutorAddingTableDirectly fails.

File: cdc/scheduler/internal/tp/agent.go [hunk 4]
```diff
@@ -94,25 +94,26 @@ func NewAgent(ctx context.Context,
...
+	log.Info("tpscheduler: owner found",
+		zap.String("ownerCaptureID", ownerCaptureID),
+		zap.String("captureID", captureID),
 		zap.String("namespace", changeFeedID.Namespace),
-		zap.String("changefeed", changeFeedID.ID),
-		zap.String("ownerID", captureID))
+		zap.String("changefeed", changeFeedID.ID))
 
-	revision, err := etcdClient.GetOwnerRevision(etcdCliCtx, captureID)
+	revision, err := etcdClient.GetOwnerRevision(etcdCliCtx, ownerCaptureID)
```
[the remark this answers is not in the document]
Comment (3AceShowHand) [reply cdc/scheduler/internal/tp/agent.go:110]: no, a mock etcd client is expected.

**Review by overvenus — APPROVED** [review]

**Pull request merged**